Language/Classes/Constructors/Factories/redirecting_to_itself_t07.dart

```diff
@@ -15,22 +18,25 @@ class F1 {
 // [analyzer] unspecified
 // [cfe] unspecified
 }
+
 typedef F1Alias = F1;
 
-class F2 {
+class F2 extends F1 {
   factory F2() = F3;
 //               ^^
 // [analyzer] unspecified
 // [cfe] unspecified
 }
 
-class F3 {
-  factory F3() = FAlias1;
+class F3 extends F2 {
+  factory F3() = F1Alias;
```

Comment:
Same issue as before: We can't assume that we're testing the error which is concerned with cyclic redirection, because `F1` (and hence `F1Alias`) isn't a subtype of `F3`.